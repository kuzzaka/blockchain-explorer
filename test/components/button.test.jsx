import React from 'react';
import { mount, shallow } from 'enzyme';
import Button from '../../components/button';

const onClick = jest.fn();

test('Button', () => {
  const wrapper = mount(<Button onClick={onClick}>click me</Button>);
  expect(wrapper.find('button')).toExist();
  wrapper.find('button').simulate('click');
  expect(onClick).toHaveBeenCalled();
});

test('Button snapshot', () => {
  const wrapper = shallow(<Button onClick={onClick}>click me</Button>);
  expect(wrapper).toMatchSnapshot();
});
