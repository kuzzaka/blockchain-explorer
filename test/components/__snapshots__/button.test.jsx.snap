// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<button
  className="jsx-905392263"
  onClick={[MockFunction]}
  type="button"
>
  click me
  <JSXStyle
    css="button.jsx-905392263{-webkit-appearance:none;-moz-appearance:none;appearance:none;font-size:inherit;padding:.5rem;cursor:pointer;border:2px solid black;min-width:8rem;}"
    styleId="905392263"
  />
</button>
`;
